login/token_src.go

```diff
@@ -224,11 +231,11 @@ func (s *OIDCTokenSource) newToken() (*oidc.Token, error) {
 		extra.Set("nonce", nonce)
 	}
 
-	ctx, cancel := context.WithTimeout(s.ctx, 1*time.Minute)
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
kill this or use given context instead background ctx

Reply:
I've replaced the background context with ctx from args. Not sure about killing context.WithCancel, you are calling cancel in goroutine after `<-quit` succeeds. Wdyt?